R/ChiSquaredDistribution.R

```diff
@@ -0,0 +1,190 @@
+#' @exportClass ZSquared
+setClass("ZSquared", contains = "ChiSquared")
+
+
+#' @param df number of degrees of freedom
+#'
+#' @examples
+#' datadist <- ChiSquared(df=4)
+#'
+#' @seealso see \code{\link{probability_density_function}} and
+#'    \code{\link{cumulative_distribution_function}} to evaluate the pdf
+#'    and the cdf, respectively.
+#'
+#' @rdname ChiSquaredDataDistribution-class
+#' @export
+ChiSquared <- function(df) {
+  if (df < 0 || abs(df - round(df)) > sqrt(.Machine$double.eps))
+    stop("The degrees of freedom must be natural numbers.")
+  new("ChiSquared", df = df, multiplier = 1)
+}
+
+
+#' Pearson's chi-squared test for 2 x k contingency tables
+#'
+#' When we test for homogeneity of rates in a k-armed trial with binary endpoints, 
+#' the test statistic is chi-squared distributed with \eqn{k-1} degrees of 
+#' freedom under the null. Under the alternative, the statistic is chi-squared 
+#' distributed with a non-centrality parameter \eqn{\lambda}. 
+#' The function \code{get_tau_Pearson2xk} then computes \eqn{\tau}, such that 
+#' \eqn{\lambda} is given as \eqn{n \cdot \tau}, where \eqn{n} is the number of 
+#' subjects per group. In \code{adoptr}, \eqn{\tau} is used in the same way as \eqn{\theta}
+#' in the case of the normally distributed test statistic.
+#'
+#' @param n_groups number of groups considered for testing procedure
+#'
+#' @examples
+#' pearson <- Pearson2xK(3)
+#'
+#'
+#' @rdname Pearson2xK-class
+#' @export
+Pearson2xK <- function(n_groups) {
+  if (n_groups < 0 || abs(n_groups - round(n_groups)) > sqrt(.Machine$double.eps))
+    stop("The number of groups must be a natural number.")
+  new("Pearson2xK", df = n_groups - 1L, multiplier = 1 / n_groups)
+}
+
+#' @param p_vector vector denoting the event rates per group
+#'
+#' @examples
+#' H1 <- PointMassPrior(get_tau_Pearson2xK(c(.3, .25, .4)), 1)
+#'
+#' @rdname Pearson2xK-class
+#'
+#' @export
+get_tau_Pearson2xK <- function(p_vector) {
+  n_groups <- length(p_vector)
+  mean_p <- mean(p_vector)
+  deltas <- p_vector - mean_p
+  tau <- (sum(deltas^2) / n_groups) / (mean_p * (1 - mean_p))
+  tau
+}
+
+
+#' Squared normally distributed random variable
+#'
+#' Implementation of \eqn{Z^2}, where \eqn{Z} is normally distributed with mean 
+#' \eqn{\mu} and variance \eqn{\sigma^2}. \eqn{Z^2} is chi-squared distributed 
+#' with \eqn{1} degree of freedom and non-centrality parameter \eqn{(\mu/\sigma)^2}.
+#' The function \code{get_tau_ZSquared} computes the factor \eqn{\tau=(\mu/\sigma)^2}, 
+#' such that \eqn{\tau} is the equivalent of \eqn{\theta} in the normally 
+#' distributed case. The square of a normal distribution \eqn{Z^2} can be used 
+#' for two-sided hypothesis testing.
+#'
+#' @param two_armed logical indicating if a two-armed trial is regarded
+#'
+#' @examples
+#' zsquared <- ZSquared(FALSE)
+#'
+#'
+#' @rdname ZSquared-class
+#' @export
+ZSquared <- function(two_armed = TRUE) {
+  new("ZSquared", df = 1L, multiplier = 1L + two_armed)
```

Comment:
Maybe it would be more clear to explicitly two_armed to an integer? (i.e. ifelse(two_armed, 1, 0))